generic.c

```diff
@@ -164,3 +164,33 @@ int createFile(const char * file, const char * buff, size_t size)
 	return SUCCESS;
 }
 
+/*
+ *returns a new pointer to a string array with one more unused index at the end 
+ *@param arr , a pointer to the string array
+ *			will be reallocated to have one more index at the end or NULL if error
+ *@param old_length , the current length of the array, the returned array will have old_length+1 memory locations
+ *@return 0 for success or ALLOC_FAIL if fail (memory will be freed in this case)
+ */
+int reallocArray(const char ***arr, int old_length)
+{
+	const char **new_arr = NULL;
+	new_arr = malloc( (old_length + 1) * sizeof(char*));
```

Comment:
Using a multiplication inside a malloc is usually considered poor practice because you can end up with integer overflow bugs. See `__builtin_mul_overflow` and friends.

I'm confused that you've re-implemented realloc from the basic primitives of malloc and free, rather than wrapping realloc to handle it's quirky return values. I would have done something more like:

```
old_arr = arr;
arr = realloc(arr, new_size);
if (arr == NULL) {
  free(old_arr);
  return ALLOC_FAILED;
}
return SUCCESS;